src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -1376,13 +1178,16 @@ private void RerouteAllQueuedMessages(ActivationData activation, ActivationAddre
 
                 activationsFailedToActivate.Increment();
 
+                /// TODO: During lifecycle refactor discuss with team whether activation failure should have a well defined exception, or throw whatever
+                ///   exception caused activation to fail, with no indication that it occured durring activation
+                ///   rather than the grain call.
+                OrleansLifecycleCanceledException canceledException = exc as OrleansLifecycleCanceledException;
+                if(canceledException?.InnerException != null)
+                {
+                    throw canceledException.InnerException;
```

Comment:
This would completely wipe out the original stack trace of the exception, which is very important for troubleshooting what went wrong. You might want to do the following instead:
```
ExceptionDispatchInfo.Capture(canceledException.InnerException).Throw();
```